_posts/2024-9-20-announcing-sprocket

```diff
@@ -0,0 +1,68 @@
+---
+title: "Sprocket"
+author: Venkat Malladi
+date:   2024-09-20 15:24:15 -0600
+categories: wdl bioinformatics workflows vscode lint
+---
+
+# Introducing Sprocket
+
+We are excited to announce **Sprocket**, a linter and validity of Workflow Description Language (WDL). Additionally, it is underactive development to become a workflow engine for bioinformatics and computational research. Sprocket is designed to be highly compatible with WDL, making it a valuable addition to the growing ecosystem of WDL tools.
+
+
+## What is Sprocket?
+
+
+Sprocket is written in Rust. The code that drives Sprocket is split across the [wdl](https://github.com/stjude-rust-labs/wdl) family of crates, the [sprocket](https://github.com/stjude-rust-labs/sprocket) command line tool, and the [Visual Studio Code extension](https://marketplace.visualstudio.com/items?itemName=stjude-rust-labs.sprocket-vscode).
+
+Sprocket aims for the following:
+
+- **WDL Compatibility**: Full support for workflows written in WDL, allowing users to define complex workflows in a readable and intuitive format.
+- **Lightweight and Fast**: Optimized for speed and minimal resource usage, Sprocket is a perfect fit for small-to-medium-sized computational projects.
+- **Flexible Deployment**: Sprocket can run on local machines, clusters, and cloud environments, giving users the freedom to scale their workflows as needed.
+- **Open-Source**: Developed under an open-source license, Sprocket encourages contributions from the community to extend its capabilities.
+
+## Why Use Sprocket?
+
+Sprocket aims at providing a minimal yet robust alternative for testing and running WDL workflows.
+
+- **Simplicity**: For users who need to run WDL workflows quickly without complex infrastructure setup, Sprocket offers a straightforward solution.
```

Comment:
This is a pretty interesting point to call out—we're specifically targeting mid to high scale bioinformatics workflow analysis. In particular, I do think there will be other solutions that are better for very small WDL analysis (e.g., miniwdl).